sdk/core/Azure.Core/src/Shared/AzureSasCredentialPolicy.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class AzureSasCredentialPolicy : HttpPipelineSynchronousPolicy
```

Comment:
We should consider deriving from the base policy class so we can eventually:

> While there's not a scenario that requires it today, we can think about supporting automatic refresh in a future release. The pipeline policy would listen for 403s on the way back, check if the SAS token expired, raise an event (probably sync or async depending on the path), allow the event to compute/fetch a new SAS token, and mark the message as retriable so customers never notice the expiration. This could be added without breaking.

Reply:
Pedantically, by REST conventions, retries should only be performed on a 401.   A 403 should be considered terminal and indicating "we know who you are and you'll never be allowed to do this."   

Do we have services not returning 401 for expired credentials?

Reply:
Since AzureSasCredentialPolicy is internal class - can this be deferred until it's actually implemented? We'd be doing [same thing](https://github.com/Azure/azure-sdk-for-net/blob/2b594335737a65cb8f36543ef817deaba5c43f4b/sdk/core/Azure.Core/src/Pipeline/HttpPipelineSynchronousPolicy.cs#L26) synchronous policy does in SAS policy.

Reply:
I think it's fine. I would just rename this class to AzureSasCredentialSynchronousPolicy. It's shared source and it might be easier to transition to a base policy by adding a new subclass and removing this one overtime.